crud/select/compat/common.lua

```diff
@@ -1,6 +1,32 @@
+local ratelimit = require('crud.ratelimit')
+local check_select_safety_rl = ratelimit.new()
+
 local common = {}
 
 common.SELECT_FUNC_NAME = '_crud.select_on_storage'
 common.DEFAULT_BATCH_SIZE = 100
 
+common.check_select_safety = function(space_name, user_conditions, opts)
+    if opts.fullscan ~= nil and opts.fullscan == true then
+        return
+    end
+
+    if opts.first ~= nil and math.abs(opts.first) <= 1000 then
+        return
+    end
+
+    if user_conditions ~= nil then
+        for _, v in ipairs(user_conditions) do
+            local it = v[1]
+            if it ~= nil and type(it) == 'string' and (it == '=' or it == '==') then
+                return
+            end
+        end
+    end
+
+    local rl = check_select_safety_rl
+    local traceback = debug.traceback()
+    rl:log_crit("Potentially long select from space '%s'\n %s", space_name, traceback)
+end
```

Comment:
> a full scan will be performed, since non-indexed field is placed first in search conditions

https://github.com/tarantool/crud/blob/52b2b4a78832ee8c97a8b946cb1764fcf59f0755/doc/select.md#avoiding-full-scan

I think it would be better to use query plan information (see `crud/compare/plan.lua`) to check, whether full scan will be performed. We need database schema information to actually say, whether particular select query will fall into the linear space scan. The query planning already uses the schema to create the plan.